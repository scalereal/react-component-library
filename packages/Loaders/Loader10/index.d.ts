import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const Loader10: React.FC<Props> & WithStyle;
export default Loader10;
