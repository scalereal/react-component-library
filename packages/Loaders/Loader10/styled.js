var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    position: relative;\n\t&:before,\n\t&:after {\n\t\tcontent: '';\n\t\tdisplay: block;\n\t\tposition: absolute;\n\t\tbackground-color: currentcolor;\n\t\tleft: 50%;\n\t\tright: 0;\n\t\ttop: 0;\n\t\tbottom: 50%;\n\t\tbox-shadow: -.5em 0 0 currentcolor;\n\t\tanimation: loader-12 1s linear infinite;\n\t}\n\t&:after {\n\t\ttop: 50%;\n\t\tbottom: 0;\n\t\tanimation-delay: .25s;\n\t}\n\n    @keyframes loader-12 {\n        0%,\n        100% {\n            box-shadow: -.5em 0 0 transparent;\n            background-color: currentcolor;\n        }\n        50% {\n            box-shadow: -.5em 0 0 currentcolor;\n            background-color: transparent;\n        }\n    }\n\n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    position: relative;\n\t&:before,\n\t&:after {\n\t\tcontent: '';\n\t\tdisplay: block;\n\t\tposition: absolute;\n\t\tbackground-color: currentcolor;\n\t\tleft: 50%;\n\t\tright: 0;\n\t\ttop: 0;\n\t\tbottom: 50%;\n\t\tbox-shadow: -.5em 0 0 currentcolor;\n\t\tanimation: loader-12 1s linear infinite;\n\t}\n\t&:after {\n\t\ttop: 50%;\n\t\tbottom: 0;\n\t\tanimation-delay: .25s;\n\t}\n\n    @keyframes loader-12 {\n        0%,\n        100% {\n            box-shadow: -.5em 0 0 transparent;\n            background-color: currentcolor;\n        }\n        50% {\n            box-shadow: -.5em 0 0 currentcolor;\n            background-color: transparent;\n        }\n    }\n\n"])), function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n           ";
        case "XS":
            return "font-size: 1rem;\n            \n            ";
        case "S":
            return "font-size: 2rem;\n            ";
        case "M":
            return "font-size: 3rem;\n            ";
        case "L":
            return "font-size: 5rem;\n            ";
        case "XL":
            return "font-size: 6rem;\n            ";
        default:
            return "font-size: 1rem;\n       ";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.visible ? 'inline-block' : 'none'; });
var templateObject_1, templateObject_2;
