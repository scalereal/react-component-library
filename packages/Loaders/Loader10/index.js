var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { Loader, LoaderWrapper } from './styled';
var Loader10 = React.memo(React.forwardRef(function (props, ref) {
    var size = props.size;
    return (React.createElement(LoaderWrapper, __assign({ ref: ref, size: size }, props),
        React.createElement(Loader, __assign({}, props))));
}));
Loader10.displayName = "Loader10";
Loader10.defaultProps = {
    size: 'S',
    visible: true
};
export default Loader10;
