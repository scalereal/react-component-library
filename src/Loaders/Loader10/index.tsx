import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const Loader10:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

Loader10.displayName="Loader10";
Loader10.defaultProps = {
    size:'S',
    visible: true
};

export default Loader10;
