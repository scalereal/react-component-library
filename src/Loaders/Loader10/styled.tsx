import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    position: relative;
	&:before,
	&:after {
		content: '';
		display: block;
		position: absolute;
		background-color: currentcolor;
		left: 50%;
		right: 0;
		top: 0;
		bottom: 50%;
		box-shadow: -.5em 0 0 currentcolor;
		animation: loader-12 1s linear infinite;
	}
	&:after {
		top: 50%;
		bottom: 0;
		animation-delay: .25s;
	}

    @keyframes loader-12 {
        0%,
        100% {
            box-shadow: -.5em 0 0 transparent;
            background-color: currentcolor;
        }
        50% {
            box-shadow: -.5em 0 0 currentcolor;
            background-color: transparent;
        }
    }

`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
           `;
        case "XS":
            return `font-size: 1rem;
            
            `;
        case "S":
            return `font-size: 2rem;
            `;
        
        case "M":
            return `font-size: 3rem;
            `;

        case "L":
            return `font-size: 5rem;
            `;
        
        case "XL":
            return `font-size: 6rem;
            `;
  
      default:
        return `font-size: 1rem;
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.display?'inline-block':'none'};
`;
