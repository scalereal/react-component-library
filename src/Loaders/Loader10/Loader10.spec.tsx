import { render } from '@testing-library/react';
import * as React from 'react';
import Loader10 from '.';

describe('Loader10 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader10/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader10
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader10 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});