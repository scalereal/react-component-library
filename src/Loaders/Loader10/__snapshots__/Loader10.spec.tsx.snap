// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader10 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz jVwPke"
  >
    <div
      class="sc-bdfBwQ duBrDA"
    />
  </div>
</div>
`;

exports[`Loader10 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz itwfOQ"
  >
    <div
      class="sc-bdfBwQ bqDvPx"
    />
  </div>
</div>
`;

exports[`Loader10 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz FEXmB"
  >
    <div
      class="sc-bdfBwQ duBrDA"
    />
  </div>
</div>
`;
